OMGVA PoS/ClientApp/src/services/taxService.ts

```diff
@@ -0,0 +1,92 @@
+﻿import axios from 'axios';
+
+export interface Tax {
+    id: string,
+    taxType: string,
+    percent: string,
+}
+export interface TaxUpdateRequest {
+    taxType?: string,
+    percent?: string,
+}
+export interface TaxCreateRequest {
+    taxType: string,
+    percent: number
+}
+
+const getAllTaxes = async (token: string | null): Promise<{ result?: Array<Tax>, error?: string }> => {
+    try {
+        const response = await axios.get('/api/tax', {
+            headers: { Authorization: `Bearer ${token}` },
+        });
+        if (response.status === 200) {
+            return { result: response.data };
+        } else {
+            return { error: response.data.message };
+        }
+    } catch (error: any) {
+        return { error: error.message || 'An unexpected error occurred.' };
+    }
+};
+
+const getTax = async (token: string | null, id: string): Promise<{ result?: Tax, error?: string }> => {
+    try {
+        const response = await axios.get(`/api/tax/${id}`, {
+            headers: { Authorization: `Bearer ${token}` },
+        });
```

Comment:
Since we will use Authorization for pretty much every endpoint, would it be possible to somehow extract the functionality of adding the header to a wrapper or something? For example `await omgvaApi.get('/tax/${id}', token)`

Reply:
It would definitely be possible. Another question if it is worth the effort.

Reply:
Even in this file the same Authorization header adding logic is used 5 times, in _businessService.ts_ it's also already used 4 times. That's 9places already for only two services and we plan to add a lot more, I think it's worth it and makes the code more readable

Reply:
Yeah, but it did not take long to copy these lines. And for readability part, I doubt there will be many readers in th =e next 10 days.